src/kvstore/kvstore_dist.h

```diff
@@ -207,6 +212,7 @@ class KVStoreDist : public KVStoreLocal {
 
       // push to servers
       size_t size = send_buf.shape().Size();
+      mkl_set_tblob_eager_mode(send_buf.data());
```

Comment:
Maybe a if case should be added there in case someone does not compile with mkl?